safety_shield/scripts/urdf_capsule_generation.py

```diff
@@ -0,0 +1,527 @@
+            f.write('\t{:.4f}, {:.4f}, {:.4f}, {:.4f}, \n'.format(T[3, 0], T[3, 1], T[3, 2], T[3, 3]))
+            f.write('\t### \n')
+        f.write(']\n')
+        f.write("enclosures: [\n")
+        for i, (p1, p2, r) in enumerate(capsules):
+            if i == 0 and ignore_first:
+                continue
+            f.write(f"{p1[0]:.8f}, {p1[1]:.8f}, {p1[2]:.8f},\n")
+            f.write(f"{p2[0]:.8f}, {p2[1]:.8f}, {p2[2]:.8f},\n")
+            f.write(f"{r:.8f}")
+            if i < len(capsules) - 1:
+                f.write(",\n###\n")
+            else:
+                f.write("\n")
+        f.write("]")
+
+
+def create_robot_file(limit_q_min, limit_q_max, limit_v, nb_joints, robot_name):
+    """
+    create the trajectory parameters yaml file with joint limits
+    :param limit_q_min: the lowest joint position
+    :param limit_q_max: the highest joint position
+    :param limit_v: the velocity limit
+    :param nb_joints: the specified number of joints for sanity checks
+    :param robot_name: name of the robot
+    :return: void
+    """
+    if len(limit_v) != nb_joints:
+        warnings.warn("Not all joints have velocity limits specified, using default values")
+        limit_v = [2] * nb_joints
+
+    if len(limit_q_min) != nb_joints or len(limit_q_max) != nb_joints:
+        warnings.warn("Not all joints have min and max positions specified, using default values")
+        limit_q_min = [-3] * nb_joints
+        limit_q_max = [3] * nb_joints
+
+    a_max_allowed = [10] * nb_joints
+    j_max_allowed = [400] * nb_joints
+    a_max_ltt = [2] * nb_joints
+    j_max_ltt = [15] * nb_joints
+    filepath = f'trajectory_parameters_{robot_name}.yaml'
+    with open(filepath, 'w') as f:
+        f.write(f'robot_name: "{robot_name}"\n\n')
+        f.write('# Failsafe information\n')
+        f.write(f'max_s_stop: 0.2\n')
+        f.write(f'q_min_allowed: {limit_q_min}\n')
+        f.write(f'q_max_allowed: {limit_q_max}\n')
+        f.write(f'v_max_allowed: {limit_v}\n')
+        f.write(f'a_max_allowed: {a_max_allowed}\n')
+        f.write(f'j_max_allowed: {j_max_allowed}\n')
+        f.write('# Maximum long-term planning acc / jerk\n')
+        f.write(f'a_max_ltt: {a_max_ltt}\n')
+        f.write(f'j_max_ltt: {j_max_ltt}\n\n')
+        f.write(f'nb_joints: {nb_joints}\n\n')
+        for i in range(1, nb_joints + 1):
+            f.write(f'q{i}: [0.0]\n')
+
+
+if __name__ == '__main__':
+    '''
+    CONCERT Chain:
+    mobile_base - J1_E_stator - L1_E - J2_E_stator - L2_E - .....   J99_E_stator - L_99_E      -          end_effector_E
+        |  fixed_J1_E  |   J1_E  |  fixed_J2_E |             ....          |  J99_E  |  end_effector_E_fixed_joint  | 
+        
+    In total   (fixed - rotating) (5 or 6 times) and two fixed joint at the end    
+    The last fixed join we skip, since ee_E has no physical extend.
+    '''
+
+    # Things to edit
```

Comment:
Make these variables command line arguments.

Reply:
Done in ba77b39d61fff8cae93f6d27702cc05d1c493452